src/main/java/de/zalando/aruha/nakadi/repository/zookeeper/ZookeeperUtils.java

```diff
@@ -0,0 +1,28 @@
+package de.zalando.aruha.nakadi.repository.zookeeper;
+
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+public class ZookeeperUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZookeeperUtils.class);
+
+    private ZookeeperUtils() {
+    }
+
+    public static <V> V runLocked(final Callable<V> callable, final InterProcessLock lock) throws Exception {
+        try {
+            lock.acquire();
+            return callable.call();
+        } finally {
+            try {
+                lock.release();
+            } catch (final Exception e) {
+                LOG.warn("Error occurred when releasing ZK lock", e);
```

Comment:
I don't think that it is better to throw that exception because:
1) If we came to `finally` block from "success"-branch then with throwing exception we will loose the result of callable.call()
2) If we came to `finally` block from "exception"-branch then with throwing exception we will loose the original exception that was thrown by callable.call()
3) I don't see how we can recover from exception in lock.release, so I don't see much sense in throwing that.